nifi-nar-bundles/nifi-standard-services/nifi-hbase-client-service-api/src/main/java/org/apache/nifi/hbase/HBaseClientService.java

```diff
@@ -42,24 +42,28 @@
               " such as hbase-site.xml and core-site.xml for kerberos, " +
               "including full paths to the files.")
             .addValidator(new ConfigFilesValidator())
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
             .build();
 
     PropertyDescriptor ZOOKEEPER_QUORUM = new PropertyDescriptor.Builder()
             .name("ZooKeeper Quorum")
             .description("Comma-separated list of ZooKeeper hosts for HBase. Required if Hadoop Configuration Files are not provided.")
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
             .build();
 
     PropertyDescriptor ZOOKEEPER_CLIENT_PORT = new PropertyDescriptor.Builder()
             .name("ZooKeeper Client Port")
             .description("The port on which ZooKeeper is accepting client connections. Required if Hadoop Configuration Files are not provided.")
             .addValidator(StandardValidators.PORT_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
             .build();
 
     PropertyDescriptor ZOOKEEPER_ZNODE_PARENT = new PropertyDescriptor.Builder()
             .name("ZooKeeper ZNode Parent")
             .description("The ZooKeeper ZNode Parent value for HBase (example: /hbase). Required if Hadoop Configuration Files are not provided.")
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
             .build();
 
     PropertyDescriptor HBASE_CLIENT_RETRIES = new PropertyDescriptor.Builder()
```

Comment:
Why did you skip the client retrieves?

Reply:
No specific reason. I was not sure it'd make sense to have different values across multiple environments but since there is no overhead to allow it, let's do it. As a general comment, I think we should start a discussion/JIRA (probably tagged for NiFi 2.0) to default the scope to Variable Registry for all the properties in the framework unless it's specifically disabled with Scope.NONE.

Reply:
Check line 202. I think you missed a call to evaluateExpressionLanguage there.

Reply:
doh... good catch